packages/wdio-cli/src/utils.ts

```diff
@@ -877,19 +877,18 @@ export async function setupTypeScript(parsedAnswers: ParsedAnswers) {
         ...(parsedAnswers.serenityAdapter ? serenityTypes : [frameworkPackage.package]),
         ...(parsedAnswers.runner === 'browser' ? ['@wdio/browser-runner'] : []),
         ...servicePackages
-            .map(service => service.package)
             .filter(service => (
                 /**
                  * given that we know that all "official" services have
                  * typescript support we only include them
                  */
-                service.startsWith('@wdio') ||
+                service.package.startsWith('@wdio') ||
                 /**
                  * also include community maintained packages with known
                  * support for TypeScript
                  */
-                COMMUNITY_PACKAGES_WITH_TS_SUPPORT.includes(service)
-            ))
+                COMMUNITY_PACKAGES_WITH_TS_SUPPORT.includes(service.package)
+            )).map(service => service.package)
```

Comment:
I'm not fully understanding why we need to do all the "code changes" here instead of or keeping it like it was or renaming stuff?

Reply:
Moving .map to after the filter makes for less iterations. In order to achieve this I have to made some small changes